### PR TITLE
Add separate typings for Melon's console

### LINKED ISSUE #112: [types] [project generator] Type console and remove "DOM" from default tsconfig.json
- We need to create global declaration types for [console](https://github.com/MelonRuntime/Melon/blob/main/projects/melon-runtime/MelonJs.JavaScript/Bindings/Tools/console.js) and remove the "DOM" declaration inside the default `tsconfig.json` generated by the command `npx melon new` (check [resource file](https://github.com/MelonRuntime/Melon/blob/main/projects/melon-runtime/MelonJS/Properties/Resources.resx) containing the project generated files information)

### PR DESCRIPTION
**Pull request by vilanz**
Fixes #112.

Melon doesn't have the same typings as the original DOM, so we're sticking to Melon's,

Not a fan of all the `any`s, but even the [DOM console typings](https://github.com/microsoft/TypeScript/blob/c6ff5f3b52d45842e5e56caf124cf5f5521c92ff/src/lib/dom.generated.d.ts#L17062) have them.

### CODE REVIEW

File: projects/melon-types/lib/types/Tools/console.d.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+type MConsole = {
+    group(...data: any[]): void;
+    error(...data: any[]): void;
+    warn(...data: any[]): void;
+    clear(): void;
+    read(): void;
```
Comment (EternalQuasar0206) [inline projects/melon-types/lib/types/Tools/console.d.ts]: `console.read()` returns a string everytime, even when empty ("")

File: projects/melon-runtime/MelonJS/Properties/Resources.resx [hunk 2]
```diff
@@ -302,7 +302,7 @@ babel/
       "types": [
         "melon-types"
       ],
-      "lib": [ "es2015", "DOM" ]
+      "lib": [ "es2015" ]
```
Comment (EternalQuasar0206) [inline projects/melon-runtime/MelonJS/Properties/Resources.resx:305]: I think it is safe to keep `es2022` instead of `es2015`

**Review by EternalQuasar0206 — CHANGES_REQUESTED** [review]

File: projects/melon-types/lib/types/Tools/console.d.ts (hunk 1 above)
Comment (vilanz) [reply projects/melon-types/lib/types/Tools/console.d.ts]: Fixed

File: projects/melon-runtime/MelonJS/Properties/Resources.resx (hunk 2 above)
Comment (vilanz) [reply projects/melon-runtime/MelonJS/Properties/Resources.resx]: Awesome

**Review by EternalQuasar0206 — APPROVED** [review]

**Pull request merged**